File: festival-web/src/main/java/cn/bdqfork/web/route/response/JsonResponseHandler.java
package cn.bdqfork.web.route.response;

import io.vertx.core.json.Json;
import io.vertx.ext.web.RoutingContext;

/**
 * @author bdq
 * @since 2020/2/12
 */
public class JsonResponseHandler extends AbstractResponseHandler {

    @Override
    protected void doHandle(RoutingContext routingContext, Object result) throws Exception {
        routingContext.response().end(Json.encodePrettily(result));
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/response/JsonResponseHandler.kt
package cn.bdqfork.kotlin.web.route.response

import io.vertx.core.json.Json
import io.vertx.ext.web.RoutingContext

/**
 * @author bdq
 * @since 2020/2/12
 */
class JsonResponseHandler : AbstractResponseHandler() {
    @Throws(Exception::class)
    override fun doHandle(routingContext: RoutingContext, result: Any) {
        routingContext.response().end(Json.encodePrettily(result))
    }
}